src/components/pages/Expense/AddExpense.js

```diff
@@ -1,121 +1,202 @@
+                <Form.Control
                   as='select'
-                  name='categoryId'
-                  value={values.categoryId}
-                  onChange={(e) => {
-                    const selectedCategory = categories.find(
-                      (category) => category._id === e.target.value
-                    );
-                    setFieldValue('categoryId', selectedCategory._id);
-                  }}
+                  name='categoryTitle'
+                  onChange={handleChange}
+                  onBlur={handleBlur}
+                  value={values.categoryTitle}
+                  isInvalid={touched.categoryTitle && errors.categoryTitle}
                 >
-                  <option value=''>Select a category</option>
-                  {categories.map((category) => (
-                    <option key={category._id} value={category._id}>
-                      {category.title}
-                    </option>
-                  ))}
-                </Field>
-                <ErrorMessage name='categoryId' component='div' />
-              </div>
-
-              <button type='submit' disabled={isSubmitting}>
-                Submit
-              </button>
-            </Form>
-          )}
-        </Formik>
-      </Modal.Body>
-      <Modal.Footer>
-        <Button variant='secondary' onClick={handleClose}>
-          Close
-        </Button>
-      </Modal.Footer>
+                  <option value={values.categoryTitle || ''}>
+                    {values.categoryTitle || 'Select a category title'}
+                  </option>
+                  {categories
+                    ?.filter(
+                      (category) => category.title !== values.categoryTitle
+                    )
+                    .map((category) => (
+                      <option key={category._id} value={category.title}>
+                        {category.title}
+                      </option>
+                    ))}
+                </Form.Control>
+                <Form.Control.Feedback type='invalid'>
+                  {errors.categoryTitle}
+                </Form.Control.Feedback>
+              </Form.Group>
+            </Modal.Body>
+            <Modal.Footer>
+              <Button variant='secondary' onClick={handleClose}>
+                Close
+              </Button>
+              <Button
+                variant='primary'
+                type='submit'
+                disabled={!dirty || isSubmitting}
+              >
+                {isSubmitting ? 'Saving...' : 'Save Changes'}
+              </Button>
+              {!create ? (
+                <Button
+                  variant='danger'
+                  disabled={isSubmitting}
+                  onClick={handleDeleteExpense}
+                  type='submit'
+                >
+                  {isSubmitting ? 'Deleting...' : 'Delete'}
+                </Button>
+              ) : (
+                ''
+              )}
```

Comment:
NB: I think this would be better
```suggestion
              { !create && (
                <Button
                  variant='danger'
                  disabled={isSubmitting}
                  onClick={handleDeleteExpense}
                  type='submit'
                >
                  {isSubmitting ? 'Deleting...' : 'Delete'}
                </Button>
              ) }
```